pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -151,13 +155,16 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		Todo,
+		PoolCreated { id: T::PoolId },
 	}
 
-	// Errors inform users that something went wrong.
 	#[pallet::error]
 	pub enum Error<T> {
-		Todo,
+		IndexOutOfBounds,
+		InvalidInput,
+		Internal,
+		UnknownPool,
+		UnAuthorized,
```

Comment:
I think the assets pallet has pretty good descriptive error names (https://docs.rs/pallet-assets/latest/src/pallet_assets/lib.rs.html#645), so I tried to re-use them where our semantics are similar. They call this one `NoPermission`, which I also like.

Reply:
Updated the error name in: [32f5988](https://github.com/KILTprotocol/kilt-node/pull/768/commits/32f5988b11ef763b7c81d918c878e2490703a8f5)